shanghai/logging.py

```diff
@@ -24,9 +25,14 @@ def set_logging_config(config):
 
 def _print_like(func):
     @functools.wraps(func)
-    def _wrap(self, *args):
+    def _wrap(self, *args, **kwargs):
         f = io.StringIO()
         print(*args, file=f)
+        if 'exc_info' in kwargs:
+            exc_info = kwargs.pop('exc_info')
+            if exc_info:
+                traceback.print_exc(file=f)
```

Comment:
I don't think anything about that, so I'm pretty sure that I'm unsure about what it does.

I haven't looked further into how `.exception` does things, and forgot about it here.